### PR TITLE
Navigate to the default account if there's only one

### PR DESCRIPTION
**Pull request by noloman**
# Checklist
- [X] I have described what this PR contains

**Choose one of the following two options:**
- - [X] This PR does not introduce major changes
- - [ ] This PR introduces major changes, and I have consulted @buresdv, @jboi or @mormaer in the *Mlem Development Matrix room*

**Choose one of the following two options:**
- - [ ] This PR does not change the UI in any way
- - [X] This PR adds new UI elements / changes the UI, and I have attached pictures or videos of the new / changed UI elements

# Pull Request Information

## About this Pull Request
If the user has only one account, as soon as the app is started, the `CommunityView` will be shown programmatically

## Screenshots and Videos

| GIF | Video |
| ------------- | ------------- |
| Simulator Screen Recording - iPhone 14 Pro - 2023-06-12 at 21 18 27 | Simulator Screen Recording - iPhone 14 Pro - 2023-06-12 at 21 18 27 |

## Additional Context
Hi all! I discovered Mlem yesterday when I was looking for alternatives to Reddit and I really liked it.
I have some experience with SwiftUI in my free time, but I'm also here both to help and learn, so please let me know if/how I can improve the code here.

Thanks for the great effort you guys are making!

### CODE REVIEW

**Review by ShadowJonathan — COMMENTED** [review]
LGTM but I'll have someone on the UI/UX side also look at this

**Comment by noloman** [comment]
Yes please do! I was hesitating about the interaction because the user can
always go back and see the list of accounts, and I’m not sure it’s the best
pattern to follow.

Thanks!

**Comment by ShadowJonathan** [comment]
A better pattern would be to have a setting/selection that indicates which account is default one ("Set account as default"), or to simply remember which account was logged in last, but for now I think that this'll scratch the most annoying itch.

Actually, could you remove the `closes` keyword for the issue? This PR addresses it, but it doesn't resolve it :)

**Review by mormaer — DISMISSED** [review]
I think this is a good addition as many users will just have the one account, it may hide how to add additional ones (a bit), but I think how accounts are managed and handled is something that needs input from design etc... on the whole I support this 👍 

Longer term I'm not sure if we'd want to move to using `NavigationStack` with a `path` argument instead, it depends how things link deep linking URLs pans out - given that's not on the horizon I say if we're happy with the UI side we should get it merged 👍

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Accounts Page.swift [hunk 1]
```diff
@@ -46,6 +52,13 @@ struct AccountsPage: View
                             }
                         }
                         .onDelete(perform: deleteAccount)
+                        .navigationDestination(isPresented: accountNavigationBinding(), destination: {
+                            CommunityView(account: accountsTracker.savedAccounts.first!, community: nil)
```
Comment (WestonHanners) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Accounts Page.swift:56]: No Force unwrapping please.

**Review by WestonHanners — CHANGES_REQUESTED** [review]
I love the idea of this PR, but we need to remove the force unwrapping.

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Accounts Page.swift (hunk 1 above)
Comment (noloman) [reply Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Accounts Page.swift]: I just changed it. Please let me know if I can do it better 👍

File: Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Accounts Page.swift [hunk 2]
```diff
@@ -53,10 +53,12 @@ struct AccountsPage: View
...
                         .navigationDestination(isPresented: accountNavigationBinding(), destination: {
-                            CommunityView(account: accountsTracker.savedAccounts.first!, community: nil)
-                                .onAppear
-                            {
-                                appState.currentActiveAccount = accountsTracker.savedAccounts.first!
+                            if let account = accountsTracker.savedAccounts.first {
+                                CommunityView(account: account, community: nil)
+                                    .onAppear
+                                {
+                                    appState.currentActiveAccount = accountsTracker.savedAccounts.first!
```
Comment (ShadowJonathan) [inline Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Accounts Page.swift:60]:
```suggestion
                                    appState.currentActiveAccount = account
```

Comment (noloman) [reply Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Accounts Page.swift]: Apologies, I didn't have enough coffee today

Comment (ShadowJonathan) [reply Mlem/Views/Tabs/Posts/Components/Post/Components/Instance and Community List/Accounts Page.swift]: no worries ^^

**Comment by buresdv** [comment]
> Longer term I'm not sure if we'd want to move to using NavigationStack with a path argument instead

This is the plan, but nobody wants to refactor the app to use `path`. I suppose it will be very difficult, I know becazse I tried and failed like four times

**Comment by buresdv** [comment]
> A better pattern would be to have a setting/selection that indicates which account is default one ("Set account as default")

This should definitely be the next step.

**Pull request merged**